types.go

```diff
@@ -304,3 +308,89 @@ type runtimeTimeProvider struct{}
 func (_ *runtimeTimeProvider) Now() time.Time {
 	return time.Now()
 }
+
+func (v *VoterEligibility) IsCurrentVoter() bool {
+	return v.currentVoter
+}
+
+func (v *VoterEligibility) IsPotentialVoter() bool {
+	return v.potentialVoter
+}
+
+func (v *VoterEligibility) SetPotentialVoter(isVoter bool) {
+	v.potentialVoter = isVoter
+}
+
+// VoterEligibility represents whether a node can currently vote,
+// and if it could potentially vote in the future.
+type VoterEligibility struct {
+	currentVoter   bool
+	potentialVoter bool
+}
+
+type VoterRegistry map[raft.ServerID]*VoterEligibility
```

Comment:
Again, the more conventional approach is to nest the map inside a struct, rather than define the type against a map directly.